assets/js/modules/analytics-4/datastore/properties.js

```diff
@@ -451,9 +453,15 @@ const baseActions = {
 	*updateSettingsForMeasurementID( measurementID ) {
 		const registry = yield commonActions.getRegistry();
 
-		registry
-			.dispatch( MODULES_ANALYTICS_4 )
-			.setMeasurementID( measurementID );
+		const existingMeasurementID = registry
+			.select( MODULES_ANALYTICS_4 )
+			.getMeasurementID();
+
+		if ( measurementID !== existingMeasurementID ) {
```

Comment:
Would you mind explaining what this change was needed for?

Reply:
Sure!

In this PR, we have used the `updateSettingsForMeasurementID()` action inside the `syncGoogleTagSettings()` action. In this particular case, the primary aim was not to update the measurement ID, but only the relevant settings (i.e. Google Tag settings).

For such cases, I think dispatching the `setMeasurementID()` action and thus resetting the measurement ID is unnecessary. Hence, I've set it to only run if the passed measurement ID is actually different from the current measurement ID.

Do you think this is valid?

I have also added a comment, however, if you think otherwise, I'll revert this change entirely.

Thanks!

Reply:
Thanks for clarifying. However, I'm still a little confused - you've said this is to avoid resetting the measurement ID, but if `measurementID === existingMeasurementID` (i.e. the case you're now avoiding), and `setMeasurementID( measurementID )` is invoked, it's just the same measurement ID anyway so should not have any practical effect. The term "reset" is a bit unclear as it makes it sound as though the setting would be changing, whereas it would simply remain the same.

That's why I've a bit unclear as to why this is needed. Unless there's a clear side effect of calling `setMeasurementID()` with the same value, I don't think we _need_ to add this condition - although, there's also nothing _wrong_ with it either, it was just a bit confusing to see due to the lack of clarity about why it's added in this PR.

Regarding unwanted side effects, I don't think there should be one, as `haveSettingsChanged()` / `hasSettingChanged()` do an equality check on the setting/saved setting, but maybe you've spotted something else. However if not, I think it should be safe to remove this condition.

Reply:
You're right. I think I was being too cautious here. Removed, thanks!